UnityProject/Assets/Scripts/Input System/InteractionV2/CoreComponents/InteractionComponentUtils.cs

```diff
@@ -35,10 +36,10 @@ public static bool ServerProcessCoordinatedInteraction<T>(T info, InteractionCoo
 	/// <returns>If validation succeeds and coordinator sends the interaciton msg, invokes
 	/// onValidationSuccess and returns STOP_PROCESSING. Otherwise returns CONTINUE_PROCESSING.</returns>
 	public static bool CoordinatedInteract<T>(T info, InteractionCoordinator<T> coordinator,
-		Action<T> onValidationSuccess)
+		Action<T> onValidationSuccess, string specificComponent = null)
```

Comment:
I notice you modified the core interaction classes, adding a string parameter to identify a specific component type. This shouldn't be necessary (and if it was, we should use an ID uint instead of a string so the message size doesn't become needlessly large), and this was a feature in place for security reasons as well - we don't want clients to be able to decide which component should fire on the server, otherwise clients would be able to have control over server logic such as interacting with things in certain ways when it should technically not be possible. 

What was the issue you were having that required this? If there are multiple components on the object which handle HandApply, you can edit the order in which they are checked by rearranging those components on the prefab. You should most likely have PaperBin above whatever other components it was conflicting with. You may want to revisit https://github.com/unitystation/unitystation/wiki/Implementing-Interactions

Reply:
This was the questionable decision I was worried about. This came about specifically because of the right click menu. The paper bin has the two handapply interactions- first being the new functionality by picking up and placing paper, and the second being Pickupable. When using the right click menu and selecting pick up, going through the list of hand applys will incorrectly select the paper bin functionality instead of picking it up. 
This will likely be a problem for some other objects in the future, so I'm open for any suggestions on what we can do about this

Reply:
Oh, I think you only added this so you could fix the right click Pickup interaction. It should be fixed in another way, the way right click pickup was implemented is not viable as you discovered (it should directly invoke the interaction logic rather than simply creating a HandApply interaction).

@fomalsd I'm open to just leaving this as is but we should create a bug so I can figure out a better way to achieve this. I can't think of a good way to do address this right now off the top of my head.

Reply:
ah, I only saw your message after I posted mine. Yep, it will be a problem for other objects so I'll just make a bug to fix it and we can leave it as is.

Reply:
Issue is: https://github.com/unitystation/unitystation/issues/2188